package/MDAnalysis/analysis/hbonds/wbridge_analysis.py

```diff
@@ -0,0 +1,813 @@
+
+    def _update_water_selection(self):
+        self._water = self.u.select_atoms(self.water_selection)
+        self.logger_debug('Size of water selection before filtering:'
+                          ' {} atoms'.format(len(self._water)))
+        if self.filter_first:
+            ns_water_selection = AtomNeighborSearch(self._water)
+            self._water = ns_water_selection.search(self._s1, 3. * self.distance)
+            self._water = ns_water_selection.search(self._s2, 3. * self.distance)
+
+        self.logger_debug("Size of water selection: {0} atoms".format(len(self._water)))
+        self._water_donors = {}
+        self._water_donors_h = {}
+        self._water_acceptors = {}
+        if not self._water:
+            logger.warning("Water selection '{0}' did not select any atoms."
+                           .format(str(self.water_selection)[:80]))
+        else:
+            self._water_donors = self._water.select_atoms(
+                'name {0}'.format(' '.join(self.donors)))
+            self._water_donors_h = {}
+            for i, d in enumerate(self._water_donors):
+                tmp = self._get_bonded_hydrogens(d)
+                if tmp:
+                    self._water_donors_h[i] = tmp
+            self.logger_debug("Water donors: {0}".format(len(self._water_donors)))
+            self.logger_debug("Water donor hydrogens: {0}".format(len(self._water_donors_h)))
+            self._water_acceptors = self._water.select_atoms(
+                'name {0}'.format(' '.join(self.acceptors)))
+            self.logger_debug("Water acceptors: {0}".format(len(self._water_acceptors)))
+
+    def run(self, **kwargs):
+        """Analyze trajectory and produce timeseries.
+
+        Stores the water bridge data per frame as
+        :attr:`WaterBridgeAnalysis.timeseries` (see there for output
+        format).
+
+        Parameters
+        ----------
+        verbose : bool (optional)
+             toggle progress meter output :class:`~MDAnalysis.lib.log.ProgressMeter`
+             [``True``]
+        debug : bool (optional)
+             enable detailed logging of debugging information; this can create
+             *very big* log files so it is disabled (``False``) by default; setting
+             `debug` toggles the debug status for :class:`WaterBridgeAnalysis`,
+             namely the value of :attr:`WaterBridgeAnalysis.debug`.
+
+        See Also
+        --------
+        :meth:`WaterBridgeAnalysis.generate_table` :
+               processing the data into a different format.
+        """
+        logger.info("WBridge analysis: starting")
+        logger.debug("WBridge analysis: donors    %r", self.donors)
+        logger.debug("WBridge analysis: acceptors %r", self.acceptors)
+        logger.debug("WBridge analysis: water bridge %r", self.water_selection)
+
+        remove_duplicates = kwargs.pop('remove_duplicates', True)  # False: old behaviour
+        if not remove_duplicates:
```

Comment:
Do you actually *ever* want people to use this? If not, fail with a `ValueError` if anyone sets the kwarg.

Reply:
@orbeckst I guess nobody probably wants to use it. I have rewritten the run() so remove_duplicates kwarg is not really in use.